### ISSUE TITLE
Stop training on nan

### ISSUE THREAD

**Issue by zhonge** [zhonge/cryodrgn#136]
`cryodrgn train_vae` should break if nan loss is hit instead of silently continuing for the specified number of epochs

**Comment by zhonge** [comment]
@vineetbansal could you add this feature? Thanks!

**Comment by lipan6461188** [comment]
The `--do-pose-sgd` option can frequently cause pose parameters to be nan. Because [`pose_optimizer.step()`](https://github.com/zhonge/cryodrgn/blob/f47871be81be2547938de7bde7d750f563afdb3f/cryodrgn/commands/train_vae.py#L451) not scaled in automatic mixed precision training. I fixed my code as below:

```python
            if do_pose_sgd and epoch >= args.pretrain:
                if args.amp:
                    optim.zero_grad()
                    scaler.step(pose_optimizer)
                    scaler.update()
                else:
                    pose_optimizer.step()
                if torch.any(torch.isnan(posetracker.rots_emb.weight[ind])) or torch.any(torch.isnan(posetracker.trans_emb.weight[ind])):
                    raise RuntimeError("NaN Found in Pose.")
```
The `scaler.step` can detect nan and [skipped to avoid corrupting the params](https://pytorch.org/docs/stable/amp.html#torch.cuda.amp.GradScaler.step). It should be noted that this style is incompatible with `apex.amp`.
